src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs

```diff
@@ -195,55 +178,110 @@ private void AnalyzeExpressionStatement(OperationAnalysisContext context)
                 }
 
                 private void AnalyzeDelegateCreationOrAnonymousFunction(OperationAnalysisContext operationAnalysisContext)
-                    => _hasDelegateCreationOrAnonymousFunction = true;
-
-                private void AnalyzeArgument(OperationAnalysisContext operationAnalysisContext)
                 {
-                    var argument = (IArgumentOperation)operationAnalysisContext.Operation;
-                    if (!_hasDelegateTypeArgument &&
-                        argument.Value.Type.IsDelegateType())
+                    _hasDelegateCreationOrAnonymousFunction = true;
+                    if (!_hasDelegateEscape)
                     {
-                        _hasDelegateTypeArgument = true;
+                        _hasDelegateEscape = !IsHandledOperationTreeShape(operationAnalysisContext.Operation);
                     }
-                }
 
-                private void AnalyzeConversion(OperationAnalysisContext operationAnalysisContext)
-                {
-                    var conversion = (IConversionOperation)operationAnalysisContext.Operation;
-                    if (!_hasConversionFromDelegateTypeOrAnonymousFunctionToNonDelegateType &&
-                        (conversion.Operand.Type.IsDelegateType() || conversion.Operand.Kind == OperationKind.AnonymousFunction) &&
-                        !conversion.Type.IsDelegateType())
+                    return;
+
+                    // Local functions.
+                    bool IsHandledOperationTreeShape(IOperation operation)
                     {
-                        _hasConversionFromDelegateTypeOrAnonymousFunctionToNonDelegateType = true;
+                        // We only handle certain operation tree shapes in flow analysis of delegate creations
+                        // and consider those as non-delegate escaping operation trees.
+                        // For the remaining unknown ones, we conservatively mark the operation to lead to
+                        // delegate escape, and corresponding bail out from flow analysis in ShouldAnalyze method below.
+
+                        // 1. Delegate creation or anonymous function variable initializer is handled.
+                        if (operation.Parent is IVariableInitializerOperation)
```

Comment:
i'd prefer examples with explanation/intution.

Reply:
Done, hope it is more clear.